tree/tree/inc/ROOT/InternalTreeUtils.hxx

```diff
@@ -61,18 +61,25 @@ struct RFriendInfo {
    */
    std::vector<std::vector<std::string>> fFriendChainSubNames;
 
-   void AddFriend(const std::string &treeName, const std::string &fileNameGlob, const std::string &alias = "");
+   /**
+    * Entries of each tree in each friend.
+    */
+   std::vector<std::vector<Long64_t>> fFriendEntries;
+
+   void AddFriend(const std::string &treeName, const std::string &fileNameGlob, const std::string &alias = "",
+                  Long64_t treeEntries = TTree::kMaxEntries);
 
-   void
-   AddFriend(const std::string &treeName, const std::vector<std::string> &fileNameGlobs, const std::string &alias = "");
+   void AddFriend(const std::string &treeName, const std::vector<std::string> &fileNameGlobs,
+                  const std::string &alias = "", const std::vector<Long64_t> &treeEntries = {});
```

Comment:
Wouldnt passing a `Long64_t treeEntries` be good enough? Now, this would apply cuts per tree of a single friend. We do not support such functionality in RDatasetSpec indeed. And hence is different from the model we stick to in the spec.

Reply:
> this would apply cuts per tree

I don't understand the comment: there are no cuts here, and RDatasetSpec does not contain any information about number of entries in the trees, main or friends, so it can't use this functionality anyway?